subprojects/language-jvm/src/testFixtures/groovy/org/gradle/api/tasks/compile/AbstractCompilerContinuousIntegrationTest.groovy

```diff
@@ -17,9 +17,15 @@
 package org.gradle.api.tasks.compile
 
 import org.gradle.launcher.continuous.Java7RequiringContinuousIntegrationTest
+import org.gradle.workers.internal.WorkerDaemonExpiration
 
 
 abstract class AbstractCompilerContinuousIntegrationTest extends Java7RequiringContinuousIntegrationTest {
+
+    def setup() {
+        executer.withArgument "-D${WorkerDaemonExpiration.DISABLE_EXPIRATION_PROPERTY_KEY}=true"
```

Comment:
We have tests that require isolated build daemons but rely on worker daemons expiration.
We have tests that don't require isolated build daemons but assert worker daemons reuse.
So I'd say the two are orthogonal.
I'll add the fixture `GradleExecuter#withWorkerDaemonsExpirationDisabled()`